packages/babel-generator/src/buffer.js

```diff
@@ -99,6 +98,19 @@ export default class Buffer {
   }
 
   endsWith(str: string): boolean {
+    // Fast path to avoid iterating over this._queue.
+    if (str.length === 1) {
+      let last;
+      if (this._queue.length > 0) {
+        const str = this._queue[0][0];
```

Comment:
Is this intentional that this variable has the same name as the argument? Looks like a potential bug to me